src/main/kotlin/net/maizegenetics/phgv2/cli/AlignAssemblies.kt

```diff
@@ -581,6 +598,60 @@ class AlignAssemblies : CliktCommand(help = "Align prepared assembly fasta files
         } catch (e: Exception) {
             myLogger.error("Error: could not execute anchorwave command. Run anchorwave manually and retry.")
         }
+
+        // Create a dot plot of the anchorwave data stored in the anchorspro file for this assembly.
+        val origFile = File(anchorsproFile)
+        // Filter out lines that start with '#', change tabs to commas, and join the rest with newline characters
+        // Kotline dataframe readDelim() does not handle tabs well, so we need to change them to commas.
+        val cleanContent = origFile.useLines { lines ->
+            lines.filterNot { it.startsWith("#") }
+                .map { it.replace("\t", ",") }
+                .joinToString("\n")
+        }
+
+        val dfAnchorWave = DataFrame.readDelim(cleanContent.reader())
+        // Plot the data, write to png file
+        val plot = plotDot(dfAnchorWave)
+        val plotFile = "${outputDir}/${justNameAsm}_dotplot.png"
+        val pathSVG = ggsave(plot, plotFile)
+        myLogger.info("Dot plot for ${justNameAsm} saved to: $pathSVG")
+
+    }
+
+    /**
+     * This function will plot the anchorwave data in a dot plot.  It has 1 required parameter,
+     * which is "data", a DataFrame containing data from the anchorwave generated anchorspro file.
+     *
+     * This is a default plot that is run when AlignAssemblies is run.  It is a dot plot
+     * showing basic alignment data in scatter plot form with the x-axis being the query
+     * sequence start position, the y-axis being the reference sequence start position, and
+     * the color of the data points based on the strand (red=forward, blue=reverse).
+     * The plot is faceted by the query and reference chromosomes.
+     *
+     * For  plots based on specific reference or query ids, users can use the plotDot function
+     * from rPHG.
+     *
+     */
+    fun plotDot(
```

Comment:
I wonder if, in the future, it would be better to place these visualization methods in some sort of utility class?

Reply:
@btmonier I initially had this in a Metrics class, but moved it back to AlignAssemblies.  This particular visualization is very specific to the anchorwave generated anchorspro file so I think it belongs here.  We could create an altered version of this in the future that has a more generic appeal, and that could go into a metric file.